.github/workflows/rust.yml

```diff
@@ -0,0 +1,53 @@
+name: Rust-DataFusion
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    container:
+      image: amd64/rust
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        id: rust-toolchain
+        with:
+          toolchain: stable
+          override: true
+          components: rustfmt, clippy
+      - name: Run
+        run: cargo fmt --manifest-path ./dask_planner/Cargo.toml --all -- --check
+
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64]
+    container:
+      image: ${{ matrix.arch }}/rust
+      env:
+        # Disable full debug symbol generation to speed up CI build and keep memory down
+        # "1" means line tables only, which is useful for panic tracebacks.
+        RUSTFLAGS: "-C debuginfo=1"
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Setup Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        id: rust-toolchain
+        with:
+          toolchain: stable
+          override: true
+          components: rustfmt, clippy
+      - name: Run clippy
+        run: |
+          cargo clippy --manifest-path ./dask_planner/Cargo.toml --all-targets --workspace -- -D warnings
+        env:
+          CARGO_HOME: "/github/home/.cargo"
+          CARGO_TARGET_DIR: "/github/home/target"
```

Comment:
It looks like [pre-commit-rust](https://github.com/doublify/pre-commit-rust) has a hook for this?

https://github.com/doublify/pre-commit-rust/blob/eeee35a89e69d5772bdee97db1a6a898467b686e/hooks.yaml#L15-L21

Let me know if that works for this; my preference would be to consolidate the Rust linting into the pre-commit hooks so that we're doing the full linting on every commit.